### ISSUE TITLE
Syntax checker tells all files are invalid

### ISSUE THREAD

**Issue by jkobus** [Halleck45/PhpMetrics#145]
```
php phpmetrics.phar ./project/src
1 [>---------------------------]file ./xxx.php is not valid and has been skipped
... doing some stuff ...
Analyzing System complexity. This will take few minutes...
Warning: array_merge_recursive() expects at least 1 parameter, 0 given in phar://Z:/workspace/phpmetrics.phar/src/Hal/Component/Bounds/Bounds.php on line 28
```

My php version:

```
Z:\workspace>php -v
PHP 5.4.21 (cli) (built: Nov 12 2013 18:10:08)
Copyright (c) 1997-2013 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2013 Zend Technologies
    with Xdebug v2.2.3, Copyright (c) 2002-2013, by Derick Rethans
    with Zend Extension Manager v6.0.0, Copyright (c) 2003-2013, by Zend Technologies
    - without datacache [not loaded]
    - with Zend Job Queue v6.0.0, Copyright (c) 2004-2013, by Zend Technologies [loaded] [not licensed] [disabled]
    - with Zend Session Clustering v6.0.0, Copyright (c) 2004-2013, by Zend Technologies [loaded] [licensed] [disabled]
    - with Zend Utils v6.0.0, Copyright (c) 2004-2013, by Zend Technologies [loaded] [licensed] [enabled]
    - without optimizerplus [not loaded]
    - without codetracing [not loaded]
    - without monitor [not loaded]
    - with Zend Page Cache v6.0.0, Copyright (c) 2004-2013, by Zend Technologies [loaded] [licensed] [disabled]
    - with Zend Monitor UI v6.0.0, Copyright (c) 2004-2013, by Zend Technologies [loaded] [licensed] [enabled]
```

PHP syntax check:

```
Z:\workspace>php -l ./xxx.php
No syntax errors detected in ./xxx.php
```

**Comment by beeare** [comment]
Same result when using current PHP 7 Release Candidate.

PHP version output:

    PHP 7.0.0RC4 (cli) (built: Sep 29 2015 17:43:04)
    Copyright (c) 1997-2015 The PHP Group
    Zend Engine v3.0.0-dev, Copyright (c) 1998-2015 Zend Technologies
        with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies

PHP syntax check output:

    D:\workspace\JobRouter\library>php -l xxx.php
    No syntax errors detected in xxx.php

**Comment by jkobus** [comment]
This happens only on windows.

**Comment by beeare** [comment]
The reason is the following code in method `isCorrect` of class `SyntaxChecker` in file `Hal\Component\File\SyntaxChecker.php`:

    if (0 >= version_compare('5.4.0', PHP_VERSION)) {
       $php = PHP_BINARY;
    }

    $output = shell_exec(sprintf('%s -l %s 2>&1', $php, escapeshellarg($filename)));

If PHP is installed in e.g. C:\Program Files (x86)\PHP_70 we run into trouble with the code above as it does not handle spaces in the PHP executable path correctly.

**Comment by jkobus** [comment]
:+1:

**Comment by woodywould** [comment]
Is there a fix/patch for this?

**Comment by Halleck45** [comment]
Hum, I cannot test on Windows, I'm only use Linux.

Do you think that the following code will work on Windows ? Can you test please ?

```php
        if (0 >= version_compare('5.4.0', PHP_VERSION)) {
           $php = PHP_BINARY;
        }

        $output = shell_exec(sprintf('"%s" -l %s 2>&1', $php, escapeshellarg($filename)));
```

**Comment by Halleck45** [comment]
Fixed by [#02fe107](https://github.com/Halleck45/PhpMetrics/commit/02fe107120567e10de2ec2ebf63b48f64f71779e)
